sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3928,3 +3928,88 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
 
   override def prettyName: String = "array_except"
 }
+
+@ExpressionDescription(
+  usage = """
+  _FUNC_(array, element) - Returns an array of appending an element to the end of an array
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 2, 3), 3);
+       [1,2,3,3]
+  """,
+  since = "3.0.0")
+case class ArrayAppend(left: Expression, right: Expression) extends BinaryExpression {
+
```

Comment:
You need to handle `nullable` and `foldable`, too.

Reply:
@yaooqinn You forgot to override `foldable`...

Reply:
the foldable here is the same as `BinaryExpression`, what about we just inherit it？

Reply:
Ur, I see. I missed this is a binary expr. Thanks.